packages/edit-site/src/components/global-styles/shadows-edit-panel.js

```diff
@@ -0,0 +1,441 @@
+				<ScreenHeader title={ selectedShadow.name } />
+				<FlexItem>
+					<Spacer marginBottom={ 0 } paddingX={ 4 }>
+						<DropdownMenu
+							icon={ moreVertical }
+							label={ __( 'Shadow options' ) }
+							toggleProps={ {
+								size: 'small',
+							} }
+						>
+							{ ( { onClose } ) =>
+								menuItems.map( ( item ) => (
+									<NavigableMenu
+										role="menu"
+										key={ item.action }
+									>
+										<Button
+											variant="tertiary"
+											onClick={ () => {
+												onMenuClick( item.action );
+												onClose();
+											} }
+											className="components-palette-edit__menu-button"
+										>
+											{ item.label }
+										</Button>
+									</NavigableMenu>
+								) )
+							}
+						</DropdownMenu>
+					</Spacer>
+				</FlexItem>
+			</HStack>
+			<div className="edit-site-global-styles-screen">
+				<ShadowsPreview shadow={ selectedShadow.shadow } />
+				<ShadowEditor
+					shadow={ selectedShadow.shadow }
+					onChange={ onShadowChange }
+				/>
+			</div>
+		</>
+	);
+}
+
+function ShadowsPreview( { shadow } ) {
+	const shadowStyle = {
+		boxShadow: shadow,
+	};
+
+	return (
+		<Spacer marginBottom={ 4 } marginTop={ -2 }>
+			<HStack
+				align="center"
+				justify="center"
+				className="edit-site-global-styles__shadow-preview-panel"
+			>
+				<div
+					className="edit-site-global-styles__shadow-preview-block"
+					style={ shadowStyle }
+				/>
+			</HStack>
+		</Spacer>
+	);
+}
+
+function ShadowEditor( { shadow, onChange } ) {
+	const shadowParts = getShadowParts( shadow );
+
+	const onChangeShadowPart = ( index, part ) => {
+		shadowParts[ index ] = part;
+		onChange( shadowParts.join( ', ' ) );
+	};
+
+	const onAddShadowPart = () => {
+		shadowParts.push( defaultShadow );
+		onChange( shadowParts.join( ', ' ) );
+	};
+
+	const onRemoveShadowPart = ( index ) => {
+		shadowParts.splice( index, 1 );
+		onChange( shadowParts.join( ', ' ) );
+	};
+
+	return (
+		<>
+			<VStack spacing={ 8 }>
+				<HStack justify="space-between">
+					<FlexItem>
+						<Subtitle level={ 3 }>{ 'Shadows' }</Subtitle>
+					</FlexItem>
+					<FlexItem>
+						<Button
+							size="small"
+							icon={ plus }
+							label={ __( 'Add shadow' ) }
+							onClick={ () => {
+								onAddShadowPart();
+							} }
+						/>
+					</FlexItem>
+				</HStack>
+			</VStack>
+			<Spacer />
+			<ItemGroup isBordered isSeparated>
+				{ shadowParts.map( ( part, index ) => (
+					<ShadowItem
+						key={ index }
+						shadow={ part }
+						onChange={ ( value ) =>
+							onChangeShadowPart( index, value )
+						}
+						canRemove={ shadowParts.length > 1 }
+						onRemove={ () => onRemoveShadowPart( index ) }
+					/>
+				) ) }
+			</ItemGroup>
+		</>
+	);
+}
+
+function ShadowItem( { shadow, onChange, canRemove, onRemove } ) {
+	const popoverProps = {
+		placement: 'left-start',
+		offset: 36,
+		shift: true,
+	};
+	const [ shadowObj, setShadowObj ] = useState(
```

Comment:
Nitpick: Using the local state may be problematic. For example, a plugin may update the pallet object directly, and our local state will keep the old shadow object. For collaborative editing, this will also create a bug, if two users have the same shadow opened, changes one user makes will not propagate because although the entity is changed the local state will not be changed.

It may be better to just remove the local state and a useMemo that computes the object using shadowStringToObject. Making the style object the only source of truth.

Reply:
Great catch. Thank you. 
Removed the local state here 719ca407ec889c5a871fde596fa7401e33e7188d